herddb-core/src/test/java/herddb/codec/RecordSerializerTest.java

```diff
@@ -79,4 +80,52 @@ private static void testTimestamp(String testCase, String timezone, long expecte
         assertEquals("failed for " + testCase + " delta is " + delta + " h, result is " + formattedResult, expectedResult, result.getTime());
     }
 
+    @Test
+    public void testSerializeWithNullAndNonNullTypes() {
```

Comment:
we are missing tests about serializing a NOT_NULL_INTEGER (and STRING...all of the types) and throw an IllegalArgumentException.

This is the key of this implementation, this way we are sure that for a NON_NULL_XXX you won't ever have a NULL.

Reply:
I think there might be a problem. Currently we return null if the object is null from the serialize method. 
public static byte[] serialize(Object v, int type) {
        if (v == null) {
            return null;
        }